tsl/src/remote/connection.c

```diff
@@ -452,7 +453,10 @@ handle_result_create(PGEventResultCreate *event)
 	entry->result = event->result;
 	/* Add entry as new head and set instance data */
 	list_insert_after(&entry->ln, &conn->results);
-	PQresultSetInstanceData(event->result, eventproc, entry);
+	if (!PQresultSetInstanceData(event->result, eventproc, entry))
```

Comment:
This will only fail if you pass in bad pointers. We should have the checks where those pointers are set/allocated instead.

If this is really what we want to do, it would be better to wrap these functions in, e.g., a MACRO, and raise the error in the wrapper instead of doing it in every place they are called.

Reply:
But this is the only place where we use that function, so it seems pointless to add a macro for that reason alone.